spirv_reflect.c

```diff
@@ -3046,60 +3080,58 @@ static SpvReflectResult ParseStaticallyUsedResources(SpvReflectPrvParser* p_pars
+    if (IsNull(used_accesses)) {
       SafeFree(p_called_functions);
       return SPV_REFLECT_RESULT_ERROR_ALLOC_FAILED;
     }
   }
-  used_variable_count = 0;
+  used_acessed_count = 0;
   for (size_t i = 0, j = 0; i < called_function_count; ++i) {
     while (p_parser->functions[j].id != p_called_functions[i]) {
       ++j;
     }
 
-    memcpy(&used_variables[used_variable_count], p_parser->functions[j].accessed_ptrs,
-           p_parser->functions[j].accessed_ptr_count * sizeof(*used_variables));
-    used_variable_count += p_parser->functions[j].accessed_ptr_count;
+    memcpy(&used_accesses[used_acessed_count], p_parser->functions[j].accessed_variables,
+           p_parser->functions[j].accessed_variable_count * sizeof(SpvReflectPrvAccessedVariable));
+    used_acessed_count += p_parser->functions[j].accessed_variable_count;
   }
   SafeFree(p_called_functions);
 
-  if (used_variable_count > 0) {
-    qsort(used_variables, used_variable_count, sizeof(*used_variables), SortCompareUint32);
+  if (used_acessed_count > 0) {
+    qsort(used_accesses, used_acessed_count, sizeof(*used_accesses), SortCompareAccessedVariable);
   }
-  used_variable_count = (uint32_t)DedupSortedUint32(used_variables, used_variable_count);
 
   // Do set intersection to find the used uniform and push constants
   size_t used_uniform_count = 0;
-  //
-  SpvReflectResult result0 = IntersectSortedUint32(used_variables, used_variable_count, uniforms, uniform_count,
-                                                   &p_entry->used_uniforms, &used_uniform_count);
+  SpvReflectResult result0 = IntersectSortedAccessedVariable(used_accesses, used_acessed_count, uniforms, uniform_count,
+                                                             &p_entry->used_uniforms, &used_uniform_count);
 
   size_t used_push_constant_count = 0;
-  //
-  SpvReflectResult result1 = IntersectSortedUint32(used_variables, used_variable_count, push_constants, push_constant_count,
-                                                   &p_entry->used_push_constants, &used_push_constant_count);
+  SpvReflectResult result1 = IntersectSortedAccessedVariable(used_accesses, used_acessed_count, push_constants, push_constant_count,
+                                                             &p_entry->used_push_constants, &used_push_constant_count);
 
-  for (uint32_t j = 0; j < p_module->descriptor_binding_count; ++j) {
-    SpvReflectDescriptorBinding* p_binding = &p_module->descriptor_bindings[j];
-    bool found = SearchSortedUint32(used_variables, used_variable_count, p_binding->spirv_id);
-    if (found) {
-      p_binding->accessed = 1;
+  for (uint32_t i = 0; i < p_module->descriptor_binding_count; ++i) {
+    SpvReflectDescriptorBinding* p_binding = &p_module->descriptor_bindings[i];
+    for (uint32_t j = 0; j < used_acessed_count; j++) {
+      if (used_accesses[j].variable_ptr == p_binding->spirv_id) {
+        p_binding->accessed = 1;
```

Comment:
the goal here is we can find all the accesses pointing to this descriptor binding variable